src/main/java/mate/academy/spring/service/impl/AuthenticationServiceImpl.java

```diff
@@ -1,27 +1,27 @@
 package mate.academy.spring.service.impl;
 
+import lombok.AllArgsConstructor;
+import mate.academy.spring.model.RoleName;
 import mate.academy.spring.model.User;
 import mate.academy.spring.service.AuthenticationService;
+import mate.academy.spring.service.RoleService;
 import mate.academy.spring.service.ShoppingCartService;
 import mate.academy.spring.service.UserService;
 import org.springframework.stereotype.Service;
 
+@AllArgsConstructor
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     private final UserService userService;
     private final ShoppingCartService shoppingCartService;
-
-    public AuthenticationServiceImpl(UserService userService,
-                                     ShoppingCartService shoppingCartService) {
-        this.userService = userService;
-        this.shoppingCartService = shoppingCartService;
-    }
+    private final RoleService roleService;
 
     @Override
     public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
+        user.getRoles().add(roleService.getRoleByName(RoleName.USER.toString()));
```

Comment:
Don't you have NPE here?
```suggestion
        user.getRoles().add(roleService.getRoleByName(RoleName.USER.name()));
```